pkg/apis/openebs.io/upgrade/v1alpha1/upgraderesult.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2018 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=upgrade
+
+// UpgradeResult forms the desired specification of
+// upgrade task's result that should be available for
+// all the upgrade tasks
+type UpgradeResult struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   UpgradeResultSpec   `json:"spec"`
+	Status UpgradeResultStatus `json:"status"`
+}
+
+// UpgradeResultSpec is the specifications of
+// upgrade result
+type UpgradeResultSpec struct {
+	// TODO: Add task-executor config spec here
```

Comment:
Let us not add config inside spec. Let it be outside.
Alternatively, the entire spec can be the config. Let us not introduce any new properties in the spec.
We might want to rename spec to config also.
Please think about all these options.